components/button/test/floating-buttons.visual-diff.js

```diff
@@ -0,0 +1,96 @@
+		});
+		await scroll(page, '#floating-buttons-short');
+		await transition;
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+	});
+
+	it('floats at bottom of page when always-float', async function() {
+		await scroll(page, '#floating-buttons-always-float-bottom', false);
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+	});
+
+	it('is correct with rtl', async function() {
+		const transition = waitForTransition(page, '#floating-buttons-buttons');
+		await page.evaluate(() => {
+			document.querySelector('#floating-buttons').shadowRoot.querySelector('#d2l-demo-snippet-toggle-dir').click();
+		});
+		await scroll(page, '#floating-buttons');
+		await transition;
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+		await page.evaluate(() => {
+			document.querySelector('#floating-buttons').shadowRoot.querySelector('#d2l-demo-snippet-toggle-dir').click();
+		});
+	});
+
+	describe('window less than min-height (500px)', () => {
+		before(async() => {
+			await page.setViewport({width: 800, height: 499, deviceScaleFactor: 2});
+		});
+
+		it('does not float', async function() {
+			await scroll(page, '#floating-buttons');
+			await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+		});
+
+		it('floats at bottom of page when always-float is true', async function() {
+			await scroll(page, '#floating-buttons-always-float-bottom', false);
+			await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+		});
+	});
+
+	const scroll = async(page, selector, alignToTop) => {
+		await page.evaluate((selector, alignToTop) => {
+			document.querySelector(selector).scrollIntoView(alignToTop);
+		}, selector, alignToTop);
+	};
+
+	const waitForTransition = (page, selector) => {
+		return page.evaluate((selector) => {
+			return new Promise((resolve) => {
+				const elem = document.querySelector(selector);
+				elem.shadowRoot.querySelector('.d2l-floating-buttons-container').addEventListener('transitionend', () => {
```

Comment:
Something to be weary of: I _think_ the event listener will get called three times since there are three different properties being transitioned.
```
transition: transform 500ms, border-top-color 500ms, background-color 500ms;
```
It might work out ok in this case since the timing is exactly the same.  I previously ran into some inconsistent screenshots with [button-icon.visual-diff.js](https://github.com/BrightspaceUI/core/blob/master/components/button/test/button-icon.visual-diff.js#L60), that I hope has been addressed there.

Reply:
Ah interesting - I hadn't considered that. In subsequent runs the screenshots have stayed consistent, but that's something to keep an eye on.